File: src/main/java/core/CommandHandler.java
package core;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import modules.AbstractModule;
import modules.GeneralModule;
import modules.azurlane.AzurLaneModule;
import modules.blindtest.BlindtestModule;
import modules.llsif.LLModule;
import modules.birthday.Birthday;
import modules.music.MusicModule;
import modules.picture.PictureModule;
import utils.BotUtils;

public class CommandHandler {

    // A map of modules mapping from module name to the map of commands
    static public Map<String, AbstractModule> moduleMap = new HashMap<>();

    // Thread pool
    static final public ExecutorService service = Executors.newFixedThreadPool(BotUtils.NB_THREAD);

    CommandHandler() {

		moduleMap.put("general", new GeneralModule());
		moduleMap.put("music", new MusicModule());
		moduleMap.put("blindtest", new BlindtestModule());
		moduleMap.put("picture", new PictureModule());
		moduleMap.put("llsif", new LLModule());
		moduleMap.put("al", new AzurLaneModule());
		moduleMap.put("birthday", new Birthday());

		// alias
		moduleMap.put("m", moduleMap.get("music"));
		moduleMap.put("b", moduleMap.get("blindtest"));
		moduleMap.put("p", moduleMap.get("picture"));
		moduleMap.put("ll", moduleMap.get("llsif"));
    }

}

File: src/main/kotlin/core/CommandHandler.kt
package core

import commands.azur_lane.AzurLaneCommandModule
import commands.birthday.BirthdayCommandModule
import commands.love_live.LoveLiveCommandModule
import internal.commands.AbstractCommandModule
import commands.music.MusicCommandModule
import commands.picture.PictureCommandModule
import utils.BotUtils.NB_THREAD
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

object CommandHandler {

    // A map of modules mapping from module name to the map of commands
    val COMMAND_MODULE_MAP: MutableMap<String, AbstractCommandModule> = HashMap()

    // Thread pool
    val SERVICE: ExecutorService = Executors.newFixedThreadPool(NB_THREAD)

    // Command modules
    val COMMAND_MODULES = setOf(AzurLaneCommandModule, BirthdayCommandModule,
            PictureCommandModule, LoveLiveCommandModule, MusicCommandModule)

    init {
        for (commandModule in COMMAND_MODULES)
            COMMAND_MODULE_MAP[commandModule.shortName] = commandModule
    }
}
